src/Microsoft.Extensions.SecretManager.Tools/Internal/MsBuildProjectReader.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.ProjectModel.Internal;
+using GracefulException = Microsoft.Extensions.SecretManager.Tools.Internal.GracefulException;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    internal class MsBuildProjectReader
+    {
+        public string ReadUserSecretId(IFileInfo fileInfo)
+        {
+            if (fileInfo == null || !fileInfo.Exists)
+            {
+                throw new GracefulException($"Could not find file '{fileInfo.PhysicalPath}'");
+            }
+
+            var projectDir = Path.GetDirectoryName(fileInfo.PhysicalPath);
+
+            var dotNetHome = Path.GetDirectoryName(new Muxer().MuxerPath);
+            var sdkResolver = new DotNetCoreSdkResolver(dotNetHome);
+            var sdk = sdkResolver.ResolveProjectSdk(projectDir);
+
+            // workaround limitations in using MSBuild API
+            Environment.SetEnvironmentVariable("MSBUILD_EXE_PATH", Path.Combine(sdk.BasePath, "MSBuild.exe"));
```

Comment:
Yeah, this one of those "project tools". It's deployed into a user project by dotnet-cli with that funky ".nuget/packages/.tools/" folder magic. 

The error is https://github.com/Microsoft/msbuild/issues/1097. I've also attempted to use a version of MSBuild that is shipped with my tool, but have found issues getting all of MSBuild dependencies into one place (https://github.com/Microsoft/msbuild/issues/1096)...which is why for now this prototype finds the MSBuild inside the .NET Core SDK.